docs/source/advanced/cmake.rst

```diff
@@ -0,0 +1,241 @@
+CMake Arguments
+===============
+
+Alpaka configures a lot of its functionality at compile time. Therefore a lot of compiler and link flags are needed, which are set by ``CMake`` arguments. The beginning of this section introduces the general Alpaca flag. The last parts of the section describe back-end specific flags.
+
+**Table of back-ends**
+
+   * :ref:`CPU Serial <cpu-serial>`
+   * :ref:`C++ Threads <cpp-threads>`
+   * :ref:`Boost Fiber <boost-fiber>`
+   * :ref:`Intel TBB <intel-tbb>`
+   * :ref:`OpenMP 2 Grid Block <openmp2-grid-block>`
+   * :ref:`OpenMP 2 Block Thread <openmp2-block-thread>`
+   * :ref:`OpenMP 5 <openmp5>`
+   * :ref:`CUDA <cuda>`
+   * :ref:`HIP <hip>`
+
+Common
+------
+
+ALPAKA_CXX_STANDARD
+  .. code-block::
+
+     C++ standard version
+
+alpaka_BUILD_EXAMPLES
+  .. code-block::
+
+     Build the examples
+
+BUILD_TESTING
+  .. code-block::
+
+     Build the testing tree
+
+ALPAKA_DEBUG
+  .. code-block::
+
+     Debug level
```

Comment:
@psychocoderHPC 
Can you please add some explanation. I heard the flag has some strange effects ;-)

Reply:
You can write it increases the verbosity of the runtime and cmake. @BenjaminW3 do you know which information will be printed with which level?

The strange behaviour https://github.com/alpaka-group/alpaka/blob/9a43612faaef5286b9601f23ac475dad27e36e90/cmake/alpakaCommon.cmake#L478 will be fixed soon by me.

Reply:
Valid values for this option are 0 to 2.
0 should be the default where there is no additional logging.
2 means that alpaka will spam the output window with every information it has. Especially pointers, sizes and other parameters of copies, kernel invocations and other operations will be printed.
1 enables some basic flow traces

Reply:
> You can write it increases the verbosity of the runtime and cmake. @BenjaminW3 do you know which information will be printed with which level?
> 
> The strange behaviour
> 
> https://github.com/alpaka-group/alpaka/blob/9a43612faaef5286b9601f23ac475dad27e36e90/cmake/alpakaCommon.cmake#L478
> will be fixed soon by me.

Side effects of ALPAKA_DEBUG will be removed with #1152

Reply:
@BenjaminW3 I have adopted your explanation and made it a little more documentation conform. Can you please check the spelling and if I have keep the meaning?